### ISSUE TITLE
VictoriaLogs limit() function uses data points instead of actual results

### ISSUE THREAD

**Issue by arseny-zinchenko** [VictoriaMetrics/VictoriaMetrics#7699]
### Describe the bug

When using a following query in VictoriaLogs:
```
{logtype=flow,environment=ops} seq("eni-0352f8c82da6aa229", "ACCEPT")
| extract "<region> <vpc_id> <az_id> <subnet_id> <instance_id> <interface_id> <flow_direction> <src_addr> <dst_addr> <src_port> <dst_port> <pkt_src_addr> <pkt_dst_addr> <pkt_src_aws_service> <pkt_dst_aws_service> <traffic_path> <packets> <bytes> <action>" keep_original_fields
  | filter
      interface_id:="eni-0352f8c82da6aa229"
      action:="ACCEPT"
      pkt_src_addr:ipv4_range("10.0.32.0/20")
  | stats by (pkt_src_addr) sum(bytes) bytes_total
  | sort by (bytes_total) desc limit 5 
  | math (bytes_total / (5m/1s)) as bytes_total
```
I'm expecting to see top 5 IP addresses.
Instead, I have 3 IPs, but 5 points on a graph:
![image](https://github.com/user-attachments/assets/8c04ca41-2358-4b41-aa60-b3766e8b0984)

### To Reproduce

Run a query with the `limit()` in Grafana Time Series visualization using VictoriaLogs data source

### Version

VictoriaLogs image: `victoriametrics/victoria-logs:v1.0.0-victorialogs`
VictoriaLogs Grafana data source: `victorialogs-datasource-v0.8.0`

**Comment by valyala** [comment]
The [`sort ... limit N` pipe](https://docs.victoriametrics.com/victorialogs/logsql/#sort-pipe) must be applied individually per every `step` in the [`/select/logsql/stats_query_range` API](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) starting from the commit 534371031e996ff6c1db54935147f3a02159d58e . This commit will be included in the next release of VictoriaLogs.

**Comment by valyala** [comment]
The `limit` must be properly applied per every `step` on the `[start ... end]` time range in queries passed to [`/select/logsql/stats_query_range` API](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) starting from the release [v1.1.0-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/). Closing the issue as fixed.

**Comment by valyala** [comment]
FYI, there was a bug in `v1.1.0-victorialogs`, which could lead to unexpected results from [instant stats queries](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-stats) if the query contains [`sort` pipe](https://docs.victoriametrics.com/victorialogs/logsql/#sort-pipe) with `limit`. The bug has been fixed in [v1.2.0-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/).

**Comment by arseny-zinchenko** [comment]
Hi, @valyala 
Maybe I'm using the `limit` in a wrong way? Have the same on all Grafana panels - a lot of IPs instead of just 10.
![image](https://github.com/user-attachments/assets/b7b51ce0-84be-4ffd-b7f2-b938057bde51)
